source/code/plugins/npmd_config_lib.rb

```diff
@@ -172,63 +170,71 @@ def self.getProcessedSubnetHash(subnetHash)
                         Logger::logWarn "Did not find subnet mask for subnet name #{ip["SubnetName"]} in hash", 2*Logger::loop
                         @@agent_ip_drops += 1
                     else
-                        _ipConfig.add_attribute("Mask", maskHash[ip["SubnetName"]])
-                        _agent.elements << _ipConfig
+                        _ipConfig["Mask"] = maskHash[ip["SubnetName"]];
                     end
+                    _agent["IPConfiguration"].push(_ipConfig);
                 end
-                if _agent.elements.empty?
+                _agents.push(_agent);
+                if _agents.empty?
                     @@agent_drops += 1
-                else
-                    _agents.elements << _agent
                 end
             end
             _agents
         end
 
         def self.createNetworkElements(networkArray, subnetIdHash)
-            _networks = REXML::Element.new("Networks")
+            _networks = Array.new
             networkArray.each do |x|
-                _network = REXML::Element.new("Network")
-                _network.add_attribute("Name", x["Name"])
+                _network = Hash.new
+                _network["Name"] = x["Name"];
+                _network["Subnet"] = Array.new
                 x["Subnets"].each do |sn|
+                    _subnet = Hash.new
                     _subnetId = subnetIdHash[sn]
                     if _subnetId.nil?
                         Logger::logWarn "Did not find subnet id for subnet name #{sn} in hash", 2*Logger::loop
                         @@network_subnet_drops += 1
                     else
-                        _snConfig = REXML::Element.new("Subnet")
-                        _snConfig.add_attribute("ID", subnetIdHash[sn])
-                        _snConfig.add_attribute("Disabled", "False")
-                        _snConfig.add_attribute("Tag", "")
-                        _network.elements << _snConfig
+                        _subnet["ID"] = subnetIdHash[sn];
+                        _subnet["Disabled"]  = ["False"] # TODO
+                        _subnet["Tag"]  = "" # TODO
                     end
+                    _network["Subnet"].push(_subnet);
                 end
-                if _network.elements.empty?
-                    @@network_drops += 1
-                else
-                    _networks.elements << _network
+                _networks.push(_network);
+                if _networks.empty?
+                    @@network_drops += 1                    
                 end
             end
             _networks
         end
 
-        def self.createActOnElements(actOnArray, subnetIdHash, xmlElemName)
-            _xmlElement = REXML::Element.new(xmlElemName)
-            actOnArray.each do |a|
+        def self.createActOnElements(elemArray, subnetIdHash)
+            _networkTestMatrix = Array.new
+            elemArray.each do |a|
                 _sSubnetId = "*"
                 _dSubnetId = "*"
```

Comment:
Can you please also check how is this processed for Windows? to make sure the rules are passed same for both.

Reply:
This is processed same both in Linux and Windows